zhmcclient/_nic.py

```diff
@@ -0,0 +1,183 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+A **NIC** object represents a single Network interface card (NIC)
+of a physical z Systems or LinuxONE computer that is in DPM mode
+(Dynamic Partition Manager mode).
+Objects of this class are not provided when the CPC is not in DPM mode.
+Each NIC represents a unique connection between a partition and
+a specific network adapter that is defined or installed on the CPC.
+
+A NIC is always contained in a Partition.
+"""
+
+from __future__ import absolute_import
+
+from ._manager import BaseManager
+from ._resource import BaseResource
+
+__all__ = ['NicManager', 'Nic']
+
+
+class NicManager(BaseManager):
+    """
+    Manager object for NICs. This manager object is scoped to the NICs
+    of a particular Partition.
+
+    Derived from :class:`~zhmcclient.BaseManager`; see there for common methods
+    and attributes.
+    """
+
+    def __init__(self, partition):
+        """
+        Parameters:
+
+          partition (:class:`~zhmcclient.Partition`):
+            Partition defining the scope for this manager object.
+        """
+        super(NicManager, self).__init__(partition)
+
+    @property
+    def partition(self):
+        """
+        :class:`~zhmcclient.Partition`: Parent object (Partition)
+        defining the scope for this manager object.
+        """
+        return self._parent
+
+    def list(self, full_properties=False):
+        """
+        List the NICs in scope of this manager object.
+
+        Parameters:
+
+          full_properties (bool):
+            Controls whether the full set of resource properties should be
+            retrieved, vs. only the short set as returned by the list
+            operation.
+
+        Returns:
+
+          : A list of :class:`~zhmcclient.Nic` objects.
+
+        Raises:
+
+          :exc:`~zhmcclient.HTTPError`
+          :exc:`~zhmcclient.ParseError`
+          :exc:`~zhmcclient.AuthError`
+          :exc:`~zhmcclient.ConnectionError`
+        """
+        if not self.partition.full_properties:
+            self.partition.pull_full_properties()
+
+        nics_res = self.partition.get_property('nic-uris')
+        nic_list = []
+        if nics_res:
+            for nic_uri in nics_res:
+                nic = Nic(self, nic_uri, {'element-uri': nic_uri})
+                if full_properties:
+                    nic.pull_full_properties()
+                nic_list.append(nic)
+        return nic_list
+
+    def create(self, properties):
+        """
+        Create and configures a NIC with the specified resource properties.
+
+        Parameters:
+
+          properties (dict): Properties for the new NIC.
+            See the section in the :term:`HMC API` about the specific HMC
+            operation and about the 'Create NIC' description of the members
+            of the passed properties dict.
+
+        Returns:
+
+          string: The resource URI of the new NIC.
```

Comment:
Depending on the outcome of the discussion what to return here that we started for the Adapter resource, we should change the return value here as well.